### ISSUE TITLE
Edge crashes on macOS with VerticalTabsAllowed := false

### ISSUE THREAD

**Issue by zakx** [MicrosoftDocs/Edge-Enterprise#598]
I am running Edge 117.0.2045.55 (arm64) on macOS 14.0. The bug also appears under macOS 13 and also in a few earlier versions of Edge.

In order to reproduce the issue,

1. Create a `com.microsoft.Edge.plist` file with the following content:

```xml
<?xml version="1.0" encoding="UTF-8"?>
<!DOCTYPE plist PUBLIC "-//Apple//DTD PLIST 1.0//EN" "http://www.apple.com/DTDs/PropertyList-1.0.dtd"\>
<plist version="1.0">
<dict>
 <key>VerticalTabsAllowed</key>
 <false/>
</dict>
</plist>
```

2. Copy that file to `/Library/Managed Preferences/com.microsoft.Edge.plist`. Create the folder if necessary.
3. If you had that file in there before, you might need to reboot before Edge picks the changes up. Make sure that the file is in the folder after reboot.
4. Start Edge and enter full screen (by clicking the green macOS fullscreen icon, by pressing Ctrl+Cmd+F, or via the menu `View > Enter Full Screen`, or by clicking a full screen button on a YouTube video for example -- it does not matter).
5. Edge will segfault immediately.

**Comment by Y-Kelly** [comment]
@zakx I've passed this on to the team to investigate.  Thanks!

**Comment by Y-Kelly** [comment]
@zakx Looks like our developers have found the root cause.  I'll follow up once the fix has been deployed.  Thanks!

**Comment by zakx** [comment]
Thank you for forwarding and the update!

**Comment by Y-Kelly** [comment]
@zakx A fix has been released to the Canary Channel (https://www.microsoft.com/en-us/edge/download/insider?form=MA13FJ), would you be able to test and see if you can still reproduce?  Version 120.0.2166.0 was released this morning.  Thanks!

**Comment by zakx** [comment]
Just tested it in my setup and can confirm that the issue is fixed with 120.0.2166.0. Thank you @Y-Kelly!
